src/lib/getVisibilityWatcher.ts

```diff
@@ -31,35 +32,30 @@ const initHiddenTime = () => {
 };
 
 const onVisibilityUpdate = (event: Event) => {
-  // If the document is 'hidden' and no previous hidden timestamp has been
-  // set, update it based on the current event data.
-  if (document.visibilityState === 'hidden' && firstHiddenTime > -1) {
-    // If the event is a 'visibilitychange' event, it means the page was
-    // visible prior to this change, so the event timestamp is the first
-    // hidden time.
-    // However, if the event is not a 'visibilitychange' event, then it must
-    // be a 'prerenderingchange' event, and the fact that the document is
-    // still 'hidden' from the above check means the tab was activated
-    // in a background state and so has always been hidden.
-    firstHiddenTime = event.type === 'visibilitychange' ? event.timeStamp : 0;
+  // Handle changes to hidden state
+  if (document.visibilityState === 'hidden') {
```

Comment:
Should we also check whether the event is a `visibilitychange` event?

I'm not sure if this library will add `onHidden()` functions before `prerenderingchange`, but maybe it's better to be explicit just in case?

Reply:
Added.